src/pudl/transform/classes.py

```diff
@@ -851,6 +851,51 @@ def replace_with_na(col: pd.Series, params: ReplaceWithNa) -> pd.Series:
 replace_with_na_multicol = multicol_transform_factory(replace_with_na)
 """A multi-column version of the :func:`nullify_outliers` function."""
 
+################################################################################
+# Spot fix missing values
+################################################################################
+
+
+class SpotFixes(TransformParams):
+    """Parameters that replace certain values with a manual corrected value."""
+
+    record_col: str | None = None
+    """The column used to identify a record."""
+    record_id: str | None = None
+    """The value in that column used to identify 1+ records to spot fix."""
+    fixes: dict[str, str | int | float | bool]
+    """A dictionary of the column to replace and the value to replace with."""
+
+
+def spot_fix_values(self, df: pd.DataFrame, params: SpotFixes) -> pd.DataFrame:
+    """Manually fix one-off singular missing values.
+
+    A fair amount of records have "" or 0 as a plant name.
+    Most of these records have no other data points in them and thus can be dropped.
+    But some of them actually have some information. Manual investigation of some
+    of these records led to some pretty easy identification of plant names. This
+    function takes a dictionary of these fixes and applies them to the dataframe.
+
+    Params:
+        df: Pre-processed, concatenated XBRL and DBF data.
+        params: an instance of :class:`SpotFixes`
+
+    Returns:
+        The same input DataFrame but with some spot fixes corrected.
+    """
+    # The default SpotFixValues() instance has no effect:
+    if params.record_id is None or params.record_col is None:
+        return df
+
+    elif params.record_col in list(df.columns):  # If record_col in df
+        for key in params.fixes:  # For each fix
+            if key in list(df.columns):
+                df.loc[df[params.record_col] == params.record_id, key] = params.fixes[
+                    key
+                ]  # Manually update value
```

Comment:
if there were many many more of these or if there were many fixes possible fixes for each set of ids, I could imagine this getting slow. But I think rejiggering it to both be more general and to do one compiled replacement is probably too much for this. I could imagine generating a index or mask with all of the possible fixes and generating an appropriately sized series to replace the values. I'm not sure it's worth the trouble for this but if it is simple to do than maybe it would be a good idea.

Reply:
Leaned on Zane's code below to do effectively just that.